source/agora/test/ValidatorRecurringEnrollment.d

```diff
@@ -63,3 +63,114 @@ unittest
     // Check if all validators in genesis are enrolled again
     assert(blocks[blocks.length - 1].header.enrollments.length == blocks[0].header.enrollments.length);
 }
+
+// Recurring enrollment with wrong `random_seed`
+unittest
+{
+    import agora.consensus.EnrollmentManager;
+    import agora.consensus.data.Enrollment;
+    import agora.consensus.data.Params;
+    import agora.consensus.data.Block;
+    import agora.common.Hash;
+    import agora.common.Config;
+    import agora.common.crypto.Key;
+    import geod24.Registry;
+    import core.stdc.time;
+    import std.exception;
+    import core.exception : AssertError;
+
+    // Will always try to enroll with PreImage at Height(0)
+    static class BadEnrollmentManager : EnrollmentManager
+    {
+        public this (string db_path, KeyPair key_pair,
+            immutable(ConsensusParams) params)
+        {
+            super(db_path, key_pair, params);
+        }
+
+        public override Enrollment createEnrollment (in Hash utxo,
+                                                Height height) @safe nothrow
```

Comment:
Use only four spaces for indentation, please.

Reply:
I feel like keeping stuff that overflow to the second line close to their original horizontal position gives a more readable result. But I'll change it to keep the codebase consistent :+1: Thanks!

Reply:
It also spends a lot more whitespace if it's near the end of the line. E.g.:

```D
public override void foo (in int par1,
                          in int par2,
                          in int par3,
                          in int par4)
```

vs:

```D
public override void foo (in int par1,
    in int par2, int par3, int par4)
```

Actually in your case since you're already using a new line, why not just do:

```
public override Enrollment createEnrollment (
    in Hash utxo, Height height )
```

Then they're grouped together nicely?

Reply:
That open parenthesis, look so sad and alone  :smile:

Reply:
It's like a more chonky egyptian brace.

Reply:
``` d
public override Enrollment createEnrollment (
    in Hash utxo, Height height 
) @safe {

}
```
How about this? With no func attribute, it looks weird tho :smile:

Reply:
It's very unusual compared to most D projects. Let's just be consistent with what we have, and later if we want to change anything we can use a code styling app to change the layout project-wide.